java/sage/media/format/FormatParserPlugin.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2015 The SageTV Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sage.media.format;
+
+
```

Comment:
Please add a description of how the plugin should be used. I'd also suggest changing the overall design for efficiency. It shouldn't have to instantiate a new class using the classloader for every file it wants to parse. I'd suggest this be a one method interface with parseFormat(File) and then add a new inner class to the plugin of something like ParsedFormat which just the 4 fields that would otherwise be returned from the method calls. Then FormatParser can hold onto a copy of it statically.

Reply:
What if I only create one static instance of the plugin in FormatParser, but leave the FormatParserPlugin interface the way that it is?

The initialize and deconstruct methods was meant to handle only having one instance of the plugin.  I think it will work the way you want.  Was there another reason to only have one parseFormat method.

Reply:
It's not thread safe if you do it that way where you're reusing the object like that w/ the existing interface....if you made it just one method (and it didn't maintain internal state, which there's really no reason to across multiple invocations) then that problem would go away.  Having a pattern with init/destroy like this for something that is not intended to maintain any long term state doesn't really seem necessary either.

Reply:
Understand.  I was trying to walk the implemetor down the path, but I get what you are saying.

Should I just expose one method in the interface that accepts a file and returns the ContainerFormat?  Than the implementation of the plug-in will handle setting the proper properties?

Or would you prefer that I make a new structure that they need to construct with all of the required data? And return that new class from the one parseFormat method.  I am just concerned the developer was going to implement it correctly.

Let me know what you think.

Reply:
Yeah, you can just have it return ContainerFormat directly....I don't foresee that changing so it's fine to consider it a stable interface. So just:

ContainerFormat parseFormat(File f);